OpenRA.Mods.Common/Widgets/Logic/Lobby/LobbyLogic.cs

```diff
@@ -791,6 +792,9 @@ void UpdatePlayerList()
 						template = editableSpectatorTemplate.Clone();
 
 					LobbyUtils.SetupEditableNameWidget(template, null, c, orderManager);
+
+					if (client.IsAdmin)
+						LobbyUtils.SetupEditableReadyWidget(template, null, client, orderManager, Map, !Map.RulesLoaded || Map.InvalidCustomRules);
```

Comment:
There is no guarantee that `Map.RulesLoaded` and `Map.InvalidCustomRules` won't change (from false to true) after this widget has been set up, so there's a high chance that you'll end with UI out of sync with the logic.